src/Dashboard/Pages/BuySell.tsx

```diff
@@ -1,142 +1,279 @@
-  let marketPrice = 0;
 
+  let marketPrice = 0;
 
   const handleBuyClick = () => {
-    setMode("BUY");
+    setMode("buy");
+
+    var buyButton = document.getElementById("buy-button")
+    var sellButton = document.getElementById("sell-button")
+    if (buyButton && sellButton) {
+      buyButton.style.backgroundColor = "blue"
+      sellButton.style.backgroundColor = "gray"
+    }
   };
 
   const handleSellClick = () => {
-    setMode("SELL");
+    setMode("sell");
+
+    var buyButton = document.getElementById("buy-button")
+    var sellButton = document.getElementById("sell-button")
+    if (buyButton && sellButton) {
+      sellButton.style.backgroundColor = "blue"
+      buyButton.style.backgroundColor = "gray"
+    }
   };
 
   const handleTickerChange = (event: { target: { value: React.SetStateAction<string>; }; }) => {
     setTicker(event.target.value);
-    getMarketPrice(ticker).then((r) => {
-      if (typeof r == 'number') {
+
+    setTickerError(false)
+    getMarketPriceHttp(ticker).then((r) => {
+      if (typeof r == "number") {
         marketPrice = r;
       } else {
         marketPrice = 0;
       }
-    })
-  }
+    });
+
+
+  };
 
-  const handleLimitPriceChange = (event: { target: { value: React.SetStateAction<string>; }; }) => {
+  const handleLimitPriceChange = (event: {
+    target: { value: React.SetStateAction<string> };
+  }) => {
     setLimitPrice(Number(event.target.value));
-  }
+  };
+
+  const handlePreviewClick = async () => {
 
-  const handleConfirmClick = () => {
-    if (ticker !== "" && numStocks !== 0 && mode !== "NONE" && limitPrice !== 0) {
-      setIsShown(true);
-    } else {
-      alert("Enter all required fields.")
+    if (ticker !== "" && numStocks !== 0 && limitPrice !== 0) {
+      setIsPreviewShown(true);
     }
+  };
 
+  const handleReset = () => {
+    //Erase the input of the input elements
+    const inputTickerSymbol = document.getElementById("input-tickerSymbol");
+    if (inputTickerSymbol) {
+      inputTickerSymbol.textContent = "";
+    }
+
+    const inputLimitPrice = document.getElementById("input-limit-price");
+    if (inputLimitPrice) {
+      inputLimitPrice.textContent = "";
+    }
+
+    const inputBuyStockAmount = document.getElementById(
+      "input-buy-stock-amount"
+    );
+    if (inputBuyStockAmount) {
+      inputBuyStockAmount.textContent = "";
+      inputBuyStockAmount.hidden = true;
+    }
+
+    const inputSellStockAmount = document.getElementById(
+      "input-sell-stock-amount"
+    );
+    if (inputSellStockAmount) {
+      inputSellStockAmount.textContent = "";
+      inputSellStockAmount.hidden = true;
+    }
+
+    //Reset the states
+    setIsPreviewShown(false);
+    setLimitPrice(0);
+    setMode("buy");
+    setStockAmount(0);
+    setTicker("");
   };
 
-  const handleMakeOrder = () => {
-    makeOrder();
+  const handleMakeOrder = async () => {
     console.log("Make Order ran");
-    handleClose();
+
+    const newOrder: Order = {
```

Comment:
We want to make sure that the parameters line up with what's in google cloud run. It doesn't line up with the current implementation of buysellorderhandler